File: rb/features.rb
# DungeonsAndDragons SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/ratelimit_feature'
require_relative 'feature/retry_feature'
require_relative 'feature/test_feature'
require_relative 'feature/timeout_feature'


module DungeonsAndDragonsFeatures
  def self.make_feature(name)
    case name
    when "base"
      DungeonsAndDragonsBaseFeature.new
    when "ratelimit"
      DungeonsAndDragonsRatelimitFeature.new
    when "retry"
      DungeonsAndDragonsRetryFeature.new
    when "test"
      DungeonsAndDragonsTestFeature.new
    when "timeout"
      DungeonsAndDragonsTimeoutFeature.new
    else
      DungeonsAndDragonsBaseFeature.new
    end
  end
end

File: py/features.py
# DungeonsAndDragons SDK feature factory

from feature.base_feature import DungeonsAndDragonsBaseFeature
from feature.test_feature import DungeonsAndDragonsTestFeature


def _make_feature(name):
    features = {
        "base": lambda: DungeonsAndDragonsBaseFeature(),
        "test": lambda: DungeonsAndDragonsTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
